### DISCUSSION TITLE
Asset check status names and icons

### DISCUSSION THREAD

**Discussion by johannkm** [dagster-io/dagster#16730] [General]
### Current check statuses

Take these four states for asset checks (ignoring in progress and skipped).

- evaluated, successful
- evaluated, unsuccessful, warning severity [`warning` in the UI]
- evaluated, unsuccessful, error severity [`failed` in the UI]
- not evaluated and the run failed [`execution failure` in the UI]

On the checks tab and sidebar, we have a name and icon for each:
![Screenshot 2023-09-22 at 4 24 51 PM](https://github.com/dagster-io/dagster/assets/22018973/1e01bfb1-4e27-4ec7-bbda-ee475b00c8c8)
![Screenshot 2023-09-22 at 4 26 43 PM](https://github.com/dagster-io/dagster/assets/22018973/83126943-cc9d-4044-822a-9de68df6364b)

On the asset node, we combine `failed` and `execution failure` counts, and we show the `failed` icon:

![Screenshot 2023-09-22 at 4 33 10 PM](https://github.com/dagster-io/dagster/assets/22018973/d0b20141-a06b-4342-b210-ba86b0a4c990)

### Should we stop combining `failed` and `execution failed` on the asset node?
I believe the reason we did this was due to spacing. One development since that is that we gave `execution failed` its own icon, which makes the combination more odd.

Maybe in cases where we're space constrained, we could prioritize showing failures and show some icon for 'see more'? Clicking on the checks already links to the checks tab

### Should we call `failed` something else?
Failed vs warning is confusing when the underlying difference is `AssetCheckSeverity.ERROR` (default) vs `AssetCheckSeverity.WARN`. We could call it `Error`:

![Screenshot 2023-09-22 at 4 42 38 PM](https://github.com/dagster-io/dagster/assets/22018973/2c730b47-b74d-40ab-8196-9a5a620da443)
![Screenshot 2023-09-22 at 4 42 06 PM](https://github.com/dagster-io/dagster/assets/22018973/dc3780d8-cc09-49c7-b655-7bfe6fc3838d)

But this is confusing compared to execution failure. It's odd that if my check raised an error, it gets the latter status.

- `Unsuccessful`?
- `Did not pass`?

Could we do something to communicate that the check evaluated, to distinguish against execution failed? Is there a better name for execution failed?

**Comment by sryza** [comment]
> Should we stop combining failed and execution failed on the asset node?

I'm comfortable with continuing to combine these. I think it's unavoidable that the space constraints mean that this is a lossy medium for displaying information, and we should be open to making compromises to get things to fit, as long as users are able to click in to get more info.

> Should we call failed something else?

I think yes. It's inconsistent that severity=WARN shows up as "Warning" but severity=ERROR, does not show up as "Error".

Maybe this is too verbose, but one option is "Failed, severity=WARN" and "Failed, severity=ERROR"

**Comment by braunjj** [comment]
What if we just went back to only reporting the results as "Passed" or "Failed" then used icons and a tooltip to show severity. In the alert policy checkboxes we can just write out the two Failed options as `Failed (WARN)` and `Failed (ERROR)` which matches the API nicely 
IMO the icons and color alone are enough to disambiguate a WARN failure from an ERROR failure. We already do this on the asset nodes.

![image](https://github.com/dagster-io/dagster/assets/2798333/578b81d3-2ae7-4e24-ad0f-f28ae2ea7c26)

**Reply by bkozura** [reply]
I think this is approach is super clear and concise.

**Reply by johannkm** [reply]
Nice, let's go with this

cc @dpeng817 and @hellendag for alert UI, and @bengotow for checks tab + sidebar

**Reply by johannkm** [reply]
One issue here @braunjj is we currently use hovering on these tags to link to runs. I think if we implement the 'launched at' column then we wouldn't need to link from it? https://dagsterlabs.slack.com/archives/C05KFG9TETB/p1694784225115959?thread_ts=1694719031.382389&cid=C05KFG9TETB

Though we do link to it from the sidebar, and its nice to be consistent...
